server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -104,6 +108,14 @@
     public BulkRequest() {
     }
 
+    public BulkRequest(@Nullable String globalIndex) {
+        this.globalIndex = globalIndex;
+    }    
+
+    /**
+     * @deprecated Types are in the process of being removed. Use {@link #BulkRequest(String)} instead
+     */
+    @Deprecated
     public BulkRequest(@Nullable String globalIndex, @Nullable String globalType) {
```

Comment:
It looks like CRUDDocumentationIT is using this deprecatd ctor still (amongst other). Since its code that is contained in the docs, should we move that to using the typeless constructor?

Reply:
I'm not sure - it's documentation showing how `index` and `type` can be associated with the BulkRequest and "inherited" by an IndexRequest objects that are missing these values.

I suppose it's a general point if we want documentation to include supported-yet-deprecated features? I can pull the docs/code if we think that's the right thing. Thoughts? cc @jtibshirani

Reply:
We've been removing documentation for deprecated features, with the understanding that users can refer to the previous version's documentation (where the feature is not deprecated).

Reply:
Thanks for the clarification, I think you are refering to https://www.elastic.co/guide/en/elasticsearch/client/java-rest/master/java-rest-high-document-bulk.html#CO102-1. I would be in favour of removing the mentioning of the type there as well and changing to the other ctor then.